### PR TITLE
Run app from Dust API as system for post-upsert hooks

### PR DESCRIPTION
**Pull request by PopDaph**
This PR adds some code to easily call a Dust app from front -> useful for both me and @fontanierh, working on post-upsert hooks. Sharing this part of my explo already as it cas be useful for him!

- Introduce a new role type for WorkspaceType.role -> for when the system is running a dust app, I think it's better to specify it's the system over using "none".
- Introduce a new `runApp` function on DustAPI (I'm sure I can do better on the error management + on the typing of the answer)
- Introduce a small lib file `runAppOnWorkspaceId` that handles getting the owner and keys from the workspace, and calls `runApp` -> not sure this is the best name or best location for the file.

### CODE REVIEW

File: front/lib/run_app.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  cloneBaseConfig,
+  DustProdActionRegistry,
+} from "@app/lib/actions_registry";
+import { planForWorkspace, prodAPICredentialsForOwner } from "@app/lib/auth";
+import { DustAPI } from "@app/lib/dust_api";
+import { Workspace } from "@app/lib/models";
+import { WorkspaceType } from "@app/types/user";
+
+export async function runAppOnWorkspaceId(
+  workspaceId: string,
```
Comment (spolu) [inline front/lib/run_app.ts:11]: You should take `owner` as an argument here, it's available ~everywhere in the code base normally so you should have it handy at your call site?

Comment (spolu) [inline front/lib/run_app.ts:11]: You should take owner as an argument here, it's available ~everywhere in the code base normally so you should have it handy at your call site?

File: front/lib/run_app.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  const owner = await _getSystemOwnerForWorkspaceId(workspaceId);
+  const prodCredentials = await prodAPICredentialsForOwner(owner);
+  const prodAPI = new DustAPI(prodCredentials);
+  const app = cloneBaseConfig(DustProdActionRegistry[appName]?.app);
+  const config = cloneBaseConfig(DustProdActionRegistry[appName]?.config);
+
+  return await prodAPI.runApp(app, config, inputs);
+}
+
+async function _getSystemOwnerForWorkspaceId(
```
Comment (spolu) [inline front/lib/run_app.ts:24]: As per above to be killed with fire.

Reason: it's part of the authorization layer which is provided by the methods in `auth.ts` and generally result in an `owner: WorkspaceType` object

File: front/lib/auth.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ const {
   NODE_ENV,
 } = process.env;
 
-export type RoleType = "admin" | "builder" | "user" | "none";
+export type RoleType = "admin" | "builder" | "user" | "none" | "system";
```
Comment (spolu) [inline front/lib/auth.ts:25]: I see the motivation will submit the review right now but need to think more about it as this is touching a pretty sensible part of the codebase :grimacing:

**Review by fontanierh — DISMISSED** [review]
LGTM for `auth.ts` and `dust_api.ts` changes ! Thank you 🙏 

Not convinced about the `run_app.ts` (not sure it is generic enough)

File: front/lib/dust_api.ts [hunk 4]
```diff
@@ -13,6 +13,25 @@ export type DustAPIErrorResponse = {
...
+export type DustAPINonStreamedResponseSuccess = {
+  run_id: string;
+  created_at: number;
+  run_type: "deploy";
+  config: DustAppConfigType;
+  status: {
+    run: "running" | "succeeded" | "errored";
+    blocks: DustAppRunBlockStatusEvent[];
+  };
+  traces: any;
```
Comment (PopDaph) [inline front/lib/dust_api.ts:30]: use unknown instead

**Review by spolu — COMMENTED** [review]
As per IRL

We keep "system" and will audit impact

We move `_getSystemOwnerForWorkspaceId` to `auth.ts` with a clear name and comments 
We kill `runAppOnWorkspaceId`

**Review by spolu — COMMENTED** [review]
LGTM Let me check the code out and audit a bit the places I can think of

File: front/lib/run_app.ts (hunk 2 above)
Comment (PopDaph) [reply front/lib/run_app.ts]: File killed with fire: 
-> function to get a system owner from a workspace id renamed and moved to auth.

File: front/lib/run_app.ts (hunk 1 above)
Comment (PopDaph) [reply front/lib/run_app.ts]: Discussed IRL -> function removed. Building an owner from workspace moved to auth, and this one does not make sense anymore, there's not much more logic to share in Henry and my job

**Review by fontanierh — APPROVED** [review]
LGTM 🙏

**Pull request merged**